crates/ty_python_semantic/src/types/class.rs

```diff
@@ -2669,6 +2669,44 @@ impl<'db> KnownClassLookupError<'db> {
     }
 }
 
+pub(crate) struct SliceLiteral {
+    pub(crate) start: Option<i32>,
+    pub(crate) stop: Option<i32>,
+    pub(crate) step: Option<i32>,
+}
+
+impl<'db> Type<'db> {
+    /// If this type represents a valid specialization of `slice`, returns a [`SliceLiteral`]
```

Comment:
Isn't there an additional qualification here? It has to represent a valid specialization of `slice`, where the specializations are types that allow us to perform slicing with statically-known numeric values (i.e. integer or boolean literals, or `None`) -- otherwise we'll just return `None`

Reply:
Ha yeah I was maybe making "valid" too load bearing there....  :sweat_smile: 

Done